### DISCUSSION TITLE
Rapidly incrementing WiFi LinkCount despite strong signal

### DISCUSSION THREAD

**Discussion by kubedzero** [arendst/Tasmota#20108] [Support request]
### PROBLEM DESCRIPTION
Rapidly incrementing WiFi LinkCount 

My hardwired smart outlet seems to frequently lose WiFi connection. Running `ping tasmotaIP` from another computer, I see 10-15 seconds of pings that succeed, and then 10-15 seconds of pings that fail (see snippet at the bottom). Looking at the Tasmota device, I can see its LED blinking during these outages. 

On the Tasmota Information page it says `RSSI 100%, -42 dBm` which suggests the WiFi is rock solid (the AP is 15ft from the Tasmota device). I've got other Tasmota devices further away from the AP that have no issues, so I'm positive the WiFi network itself is OK (though maybe it's a Channel 1 interference issue? I had been having issues with Channel 6 in the past) 

Anyway, the real interesting bit that I can't find any discussion about is the StatusSTS LinkCount output in the Console. A snippet is `"Wifi":{"AP":1,"SSId":"MyWifi","BSSId":"60:D0:2C:FF:FF:FF","Channel":1,"Mode":"11n","RSSI":100,"Signal":-41,"LinkCount":113,"Downtime":"0T00:03:12"}}}`. Note that the LinkCount is 113, which I think means that in the past 40 minutes of uptime it's disconnected and reconnected 113 times, with 3:12 of "downtime." I can't find any description of what these values mean, but they are fairly obvious from their names. In a couple hours of running, the LinkCount is now up at 368.

One other note is that in the logs i sometimes see `Connecting to AP1 MyWifi Channel 1 BSSId 60:D0:2C:ff:ff:ff in mode 11n as tasmota-ffffff-0242...` followed by a `Connected` line, with no other wifi-related lines. 

If it's not a Tasmota issue, could it be that the hardware is failing? I've had this running for around five years. My Tasmota version is `13.2.0.2`

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Oittm WS01 https://templates.blakadder.com/oittm_WS01.html
- [x] Tasmota binary firmware version number used: 13.2.0.2
  - [x] Pre-compiled
  - [ ] Self-compiled
- [x] Flashing tools used: OTA
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
12:12:14.971 CMD: Backlog Template; Module; GPIO 255
12:12:14.973 SRC: WebConsole from 10.1.1.27
12:12:14.975 CMD: Grp 0, Cmd 'BACKLOG', Idx 1, Len 26, Pld -99, Data 'Template; Module; GPIO 255'
12:12:14.996 SRC: Backlog
12:12:14.998 CMD: Grp 0, Cmd 'TEMPLATE', Idx 1, Len 0, Pld -99, Data ''
12:12:14.003 RSL: RESULT = {"NAME":"Oittm WS01","GPIO":[32,0,0,0,0,2592,0,0,224,2656,2688,288,0,0],"FLAG":0,"BASE":6}
12:12:15.209 SRC: Backlog
12:12:15.212 CMD: Grp 0, Cmd 'MODULE', Idx 1, Len 0, Pld -99, Data ''
12:12:15.215 RSL: RESULT = {"Module":{"0":"Oittm WS01"}}
12:12:15.459 SRC: Backlog
12:12:15.462 CMD: Grp 0, Cmd 'GPIO', Idx 1, Len 3, Pld 255, Data '255'
12:12:15.470 RSL: RESULT = {"GPIO0":{"32":"Button1"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"0":"None"},"GPIO5":{"2592":"HLWBL SEL"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"224":"Relay1"},"GPIO13":{"2656":"HLWBL CF1"},"GPIO14":{"2688":"HLW8012 CF"},"GPIO15":{"288":"Led1"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}


```
- [x] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:

- [x] Provide the output of this command: `Status 0`:
```lua
12:13:14.986 CMD: status 0
12:13:14.988 SRC: WebConsole from 10.1.1.27
12:13:14.991 CMD: Grp 0, Cmd 'STATUS', Idx 1, Len 1, Pld 0, Data '0'
12:13:14.996 RSL: STATUS = {"Status":{"Module":0,"DeviceName":"WS01-TV","FriendlyName":["WS01-TV"],"Topic":"tasmota_2C60F2","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
12:13:14.002 RSL: STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"Software/System restart","Uptime":"0T02:21:09","StartupUTC":"2023-11-25T14:52:05","Sleep":50,"CfgHolder":4617,"BootCount":8,"BCResetTime":"2023-11-25T15:02:37","SaveCount":52,"SaveAddress":"F8000"}}
12:13:14.006 RSL: STATUS2 = {"StatusFWR":{"Version":"13.2.0.2(tasmota)","BuildDateTime":"2023-11-23T11:02:28","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"333/699"}}
12:13:14.010 RSL: STATUS3 = {"StatusLOG":{"SerialLog":0,"WebLog":3,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["MyWifi",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008001","2805C80014000600003C5A0A192800000000","00000080","00006000","00004000","00000000"]}}
12:13:14.020 RSL: STATUS4 = {"StatusMEM":{"ProgramSize":636,"Free":364,"Heap":24,"ProgramFlashSize":1024,"FlashSize":1024,"FlashChipId":"146085","FlashFrequency":40,"FlashMode":"DOUT","Features":["0809","8F9AC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","54000020","00000080","00000000"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,24,26,27,29,30,35,37,45,62,68","Sensors":"1,2,3,4,5,6","I2CDriver":"7"}}
12:13:14.025 RSL: STATUS5 = {"StatusNET":{"Hostname":"tasmota-2C60F2-0242","IPAddress":"10.1.1.46","Gateway":"10.1.1.222","Subnetmask":"255.255.255.0","DNSServer1":"10.1.1.222","DNSServer2":"0.0.0.0","Mac":"60:01:94:2C:60:F2","Webserver":2,"HTTP_API":1,"WifiConfig":5,"WifiPower":17.0}}
12:13:14.030 RSL: STATUS7 = {"StatusTIM":{"UTC":"2023-11-25T17:13:14","Local":"2023-11-25T12:13:14","StartDST":"2023-03-12T02:00:00","EndDST":"2023-11-05T02:00:00","Timezone":99,"Sunrise":"02:13","Sunset":"10:59"}}
12:13:14.033 RSL: STATUS9 = {"StatusPTH":{"PowerDelta":[0,0,0],"PowerLow":0,"PowerHigh":0,"VoltageLow":0,"VoltageHigh":0,"CurrentLow":0,"CurrentHigh":0}}
12:13:14.041 RSL: STATUS10 = {"StatusSNS":{"Time":"2023-11-25T12:13:14","ENERGY":{"TotalStartTime":"2023-11-25T15:44:45","Total":1272.081,"Yesterday":0.647,"Today":0.081,"Power":0,"ApparentPower":0,"ReactivePower":0,"Factor":0.00,"Voltage":0,"Current":0.000}}}
12:13:14.047 RSL: STATUS11 = {"StatusSTS":{"Time":"2023-11-25T12:13:14","Uptime":"0T02:21:09","UptimeSec":8469,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER":"OFF","Wifi":{"AP":1,"SSId":"MyWifi","BSSId":"60:D0:2C:7B:ff:ff","Channel":1,"Mode":"11n","RSSI":100,"Signal":-42,"LinkCount":351,"Downtime":"0T00:09:39"}}}


```
- [x] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
10:15:15.003 RSL: STATUS11 = {"StatusSTS":{............
10:15:29.121 WIF: Connecting to AP1 MyWifi Channel 1 BSSId 60:D0:2C:7B:ff:ff in mode 11n as tasmota-2C60F2-0242...
10:15:30.758 WIF: Connected
10:16:14.724 RSL: STATUS = {"Status":{"Module":0,"Devic.............

```

### TO REPRODUCE
It only happens with this one device, and simply powering it on is enough. I've tried to reset it using `Reset 2` and 1 and 3 and the behavior did not change

### EXPECTED BEHAVIOUR
The LinkCount should remain low. Other devices on my network have a LinkCount of 0 or 1

### SCREENSHOTS
N/A

### ADDITIONAL CONTEXT

```
64 bytes from 10.1.1.46: icmp_seq=5624 ttl=255 time=6.261 ms
64 bytes from 10.1.1.46: icmp_seq=5625 ttl=255 time=5.966 ms
64 bytes from 10.1.1.46: icmp_seq=5626 ttl=255 time=6.774 ms
Request timeout for icmp_seq 5627
Request timeout for icmp_seq 5628
Request timeout for icmp_seq 5629
Request timeout for icmp_seq 5630
Request timeout for icmp_seq 5631
Request timeout for icmp_seq 5632
Request timeout for icmp_seq 5633
Request timeout for icmp_seq 5634
Request timeout for icmp_seq 5635
Request timeout for icmp_seq 5636
64 bytes from 10.1.1.46: icmp_seq=5637 ttl=255 time=252.087 ms
64 bytes from 10.1.1.46: icmp_seq=5638 ttl=255 time=5.839 ms
64 bytes from 10.1.1.46: icmp_seq=5639 ttl=255 time=11.992 ms
64 bytes from 10.1.1.46: icmp_seq=5640 ttl=255 time=14.394 ms
64 bytes from 10.1.1.46: icmp_seq=5641 ttl=255 time=10.308 ms
64 bytes from 10.1.1.46: icmp_seq=5642 ttl=255 time=7.461 ms
64 bytes from 10.1.1.46: icmp_seq=5643 ttl=255 time=13.702 ms
64 bytes from 10.1.1.46: icmp_seq=5644 ttl=255 time=6.283 ms
64 bytes from 10.1.1.46: icmp_seq=5645 ttl=255 time=6.367 ms
64 bytes from 10.1.1.46: icmp_seq=5646 ttl=255 time=8.475 ms
Request timeout for icmp_seq 5647
Request timeout for icmp_seq 5648
Request timeout for icmp_seq 5649
Request timeout for icmp_seq 5650
Request timeout for icmp_seq 5651
Request timeout for icmp_seq 5652
Request timeout for icmp_seq 5653
Request timeout for icmp_seq 5654
Request timeout for icmp_seq 5655
Request timeout for icmp_seq 5656
64 bytes from 10.1.1.46: icmp_seq=5657 ttl=255 time=129.937 ms
64 bytes from 10.1.1.46: icmp_seq=5658 ttl=255 time=11.671 ms
64 bytes from 10.1.1.46: icmp_seq=5659 ttl=255 time=12.412 ms
64 bytes from 10.1.1.46: icmp_seq=5660 ttl=255 time=7.620 ms
64 bytes from 10.1.1.46: icmp_seq=5661 ttl=255 time=5.324 ms
```

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by kubedzero** [comment]
I've seen a couple reboots as well with Exception as the Reason, here are the log outputs from those. Not positive if they're related, I couldn't find a guide on how to read these RestartReasons or Exceptions or Stacks

```
00:00:03.033 RSL: INFO3 = {"Info3":{"RestartReason":{"Exception":0,"Reason":"Exception","EPC":["402446ce","00000000","00000000"],"EXCVADDR":"00000000","DEPC":"00000000","CallChain":["40244695","40244734","4022b2b8","4022b2fa","4022b320","4022b33d","40238110","40101c5b","40244b25","4024f674","40223050","402530a2","40253100","401000e1","40223088","4024f11d","4022a71e","401012c2","4022a837","40254280","4022aa06","4022aa59","401012c2","40220790","40222c03","402404f0","4024059c","40100b38","402502c4","40101f51"]},"BootCount":12}}


22:23:18.868 RSL: STATUS12 = {"StatusSTK":{"Exception":0,"Reason":"Exception","EPC":["402446ce","00000000","00000000"],"EXCVADDR":"00000000","DEPC":"00000000","CallChain":["40244695","40244734","4022b2b8","4022b2fa","4022b320","4022b33d","40238110","40101c5b","40244b25","4024f674","40223050","402530a2","40253100","401000e1","40223088","4024f11d","4022a71e","401012c2","4022a837","40254280","4022aa06","4022aa59","401012c2","40220790","40222c03","402404f0","4024059c","40100b38","402502c4","40101f51"]}}
```

**Comment by Jason2866** [comment]
What device it is? The issues with wifi you see are typical when the esp has powering issues.
Another symptom for this are the exceptions you see. So when it is the only device which such behaviour it is probably a hardware issue. Replace the device and see if the problems go away.

**Comment by kubedzero** [comment]
I saw another reboot 

```
10:41:07.983 RSL: STATUS11 = {"StatusSTS":{"Time":"2023-11-26T10:41:07","Uptime":"0T01:56:30","UptimeSec":6990,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER":"OFF","Wifi":{"AP":1,"SSId":"MyWifi","BSSId":"60:D0:2C:7B:ff:ff","Channel":1,"Mode":"11n","RSSI":100,"Signal":-43,"LinkCount":326,"Downtime":"0T00:10:18"}}}
10:41:07.995 RSL: STATUS12 = {"StatusSTK":{"Exception":0,"Reason":"Exception","EPC":["4025aa42","00000000","00000000"],"EXCVADDR":"0000019d","DEPC":"00000000","CallChain":["4025aa35","4025aae8","40260f8c","4025d5d8","4026104c","402594e2","40105592","4025cdc8","40104ee1","4025ce1c","4025ea70","4025f280","40106305","4025d194","40101c92","40260890","40101c5b","4027257c","40259935","4025975b","4027ca3a","4027ca4b","402721a7","4027824f","40000f49","40000f49","40000e19","40105b65","40105b6b","40250229","4010000d"]}}
```

> What device it is?

I answered using the above template, but I'll share again https://templates.blakadder.com/oittm_WS01.html

> Another symptom for this are the exceptions you see


Can you be more specific @Jason2866? What in the exceptions indicate a powering issue? I'm trying to learn to diagnose the exact issue. Throwing up my hands and saying "bad hardware, better replace it" is a last resort. 

> Replace the device and see if the problems go away

I don't want to replace it unless absolutely necessary. I'd love to get some help translating the exceptions or behavior, or getting more pointers on troubleshooting.

**Comment by Jason2866** [comment]
CPU is powered badly -> errors occur caused by this -> ends in not reproducable errors and exceptions.
If you want to fix. Get the scope out and check the 3.3V power supply from the MCU.
Software can't fix Hardware issues ;-)

**Reply by Jason2866** [reply]
Most of the time the caps of the power supply are the components to blame. Replace this parts and with a bit of luck the device is back to working state.

**Comment by kubedzero** [comment]
> CPU is powered badly

Are you sure that could cause the 10 seconds pingable, 10 seconds not pingable behavior? To me that repeated behavior seems more of a software issue, which is why I opened an issue and not a support discussion.

If there is a power issue, i there a way to fix it? I've baked graphics cards in the past, is fracturing solder and reflowing a common practice for ESP devices? If you have other discussions/issues that you can point to where similar issues happened, I'd appreciate the links to reference. 

Edit: I see https://github.com/arendst/Tasmota/discussions/19322 discusses there being issues after Tasmota 12.2.0, and I'm wondering if a similar issue is happening here. Would reverting to an old version get this back to a working state?
